internal/provisioner/nginx_internal.go

```diff
@@ -95,15 +95,10 @@ func (n *nginxInternal) ActualVersion() *model.HelmUtilityVersion {
 }
 
 func (n *nginxInternal) Destroy() error {
-	helm := newHelmDeployment(
-		chartNameNginxInternal,
-		chartDeploymentNameNginxInternal,
-		namespaceNginxInternal,
-		n.kubeconfigPath,
-		nil,
-		defaultHelmDeploymentSetArgument,
-		n.logger,
-	)
+	helm, err := n.NewHelmDeployment()
```

Comment:
Why are we adding this? `NewHelmDeployment` does some extra tasks. I think we do not need that to delete.

Reply:
I wanted to be sure that we use the same helm deployment in both creation and deletion, this was the only provisioner that used a different one in the two methods.

Reply:
@mirshahriar Take a look at the changes